### ISSUE TITLE
Odd issue with OUI data / node vendor inventory report

### ISSUE THREAD

**Issue by kingtrw** [netdisco/netdisco#1192]
Version 2.72.1

Node vendor inventory report has very wrong values

(Unknown Vendor)	22,446
Apple, Inc.	4,434
Intel Corporate	2,530
Dell Inc.	1,551
...

the link for the (Unknown Vendor) item is to /report/nodevendor?vendor=blank which just refreshes the report.

Lansweeper agrees that we have about 1.5K Dell systems on the network, but fewer than 20 apple systems, and about the same for Intel.

**Comment by ollyg** [comment]
I'm really sorry @rc9000 and @kingtrw I can't replicate this on my data :-(

**Comment by rc9000** [comment]
I had a look. Netdisco is probably ok, it's just that we have thousands of MACs on the network that are not registered in any of the OUI databases. The left join manufacturer.base = me.oui fails for about 100k records in our case, I went and checked their range match:
```

with c as (
SELECT substr(me.mac::text, 1, 8) pref, me.mac, me.oui, manufacturer.abbrev, manufacturer.company, 
    ('x' || lpad( translate( mac::text, ':', ''), 16, '0')) ::bit(64) ::bigint as voodoo
  FROM node me
  LEFT JOIN manufacturer manufacturer
    ON manufacturer.base = me.oui
WHERE me.active and manufacturer.abbrev is null)
select pref, count(*), array_agg(mac)  from c
group by 1 order by 2 desc;
```

Results:
```

/*
-- most popular non-existent prefixes
d2:39:ea    916
02:a0:98    562
00:50:c2    377
02:08:20    125
92:e2:ba    108
86:06:47    105
86:16:0c    104
*/
```
Examples
```

/*
-- first example for each from the array_agg
d2:39:ea:13:65:ad
02:a0:98:7c:ff:1e
00:50:c2:09:72:5c -- this one is found on https://www.wireshark.org/tools/oui-lookup.html: IMV Invertomatic 00:50:C2:09:70:00/36
02:08:20:00:3a:aa -- all others are also not found on wireshark or https://www.macvendorlookup.com/
92:e2:ba:a9:05:20
*/
```

According to https://www.mist.com/get-to-know-mac-address-randomization-in-2020/, all of these are randomized addresses, so work as designed. I did a quick test with, randomized mac addresses have the "Local" bit set which translates to the second character being 2,6,a or e. This modified report would show them in a separate category while still showing other unknowns as proper null:

```
SELECT manufacturer.abbrev, 
    case 
        when manufacturer.company is null and  substr(me.mac::text, 2, 1) in ('2', '6', 'a', 'e') then 'Local Randomized Address'
        when manufacturer.company is null then null
        else manufacturer.company 
    end as company,
    COUNT (DISTINCT( me.mac ))
  FROM node me
  LEFT JOIN manufacturer manufacturer
    ON manufacturer.base = me.oui
WHERE me.active
GROUP BY manufacturer.abbrev, 2
ORDER BY count desc;
```


This gives:
```

|abbrev        |company                 |count |
|--------------|------------------------|------|
|              |Local Randomized Address|152808|
|IntelCorporate|Intel Corporate         |94367 |
|Hp            |HP Inc.                 |74357 |
|Vmware        |VMware, Inc.            |40540 |
|HewlettPackard|Hewlett Packard         |23093 |

```

I don't have time to fight the ORM right now, but I can give it a try in a bit if nobody does it before me :)

**Comment by rc9000** [comment]
Actually instead of fixing a single report, we could import the random address ranges as 64x 8-bit base manufacturer entries, then every part of Netdisco would support them. E.g. the node info in the interface list could then look like:

```
60:dd:8e:64:xx:xx (on vlan 912) (Vendor: IntelCorporate )
60:dd:8e:64:xx:xx (on vlan 912) (Vendor: IntelCorporate )
6e:23:84:2b:xx:xx (on vlan 912) (Vendor: randomized )
8c:1d:96:66:xx:xx (on vlan 912) (Vendor: IntelCorporate )
8c:7a:aa:5f:xx:xx (on vlan 912) (Vendor: Apple )
8e:0d:bc:24:xx:xx (on vlan 912) (Vendor: randomized )
96:43:07:74:xx:xx (on vlan 912) (Vendor: randomized )
```
See attached PR. One problem is that some 02 and aa ranges were assigned before the local administered bit was fleshed out. They can't be both in the DB, so I skip importing them from the oui DB and prefer the randomized range instead. See the comments in the script for details.

**Comment by ollyg** [comment]
I've merged #1198 many thanks!
